pkg/agent/controller/agent.go

```diff
@@ -564,3 +579,32 @@ func (a *Controller) filterLocalEndpointSlices(obj runtime.Object, numRequeues i
 
 	return obj, false
 }
+
+func (a *Controller) getGlobalIP(service *corev1.Service) (ip, reason, msg string) {
+	if a.globalnetEnabled {
+		ip = a.getGlobalIPFromService(service)
+		if ip != "" {
+			return ip, "", ""
+		}
+
+		ingressIP := a.getIngressIP(service.Name, service.Namespace)
+		if ingressIP == nil || ingressIP.allocatedIP == "" {
+			reason, msg = getReasonAndMsgFromIngressIP(ingressIP)
+			return "", reason, msg
+		}
```

Comment:
```suggestion
		ingressIP, notFound, err := a.getIngressIP(service.Name, service.Namespace)
		if err != nil {
		    return "", "GlobalIngressIPRetrievalFailed", err.Error()
		}
		
		if notFound {
		    return "",  defaultReasonIPUnavailable, defaultMsgIPUnavailable
		}
```

Reply:
There will be delay in creation of svcex and globalingress. We'll be requeued 3-4 times and this can lead to unnecessary padding of status to ServiceExport. Besides, currently we don't have v2 enabled. This will result in SvcEx showing status `GlobalIngressIPRetrievalFailed` when we're not even using v2.